### ISSUE TITLE
feat request: preview a staged hugo web stack in PRs

### ISSUE THREAD

**Issue by metral** [pulumi/docs#1332]
For doc and blog writers, it'd be great if we could see and share with others a preview of a staged hugo web stack when a PR is opened. This would help the reviewers and users see the changes being proposed from the PR itself without having to pull down the PR and run hugo locally.

**Comment by joeduffy** [comment]
This would indeed be very nice.

My ideal wishlist here is to have a Netlify-like experience, but using Pulumi:

* Upon a new PR being opened
    - Stand up a fresh, new static website, being served out of an S3 bucket
    - Post back that URL using our GitHub App, as an output property of the Pulumi stack
* Upon a PR being merged
    - Destroy and remove the stack

If we do this using S3 websites, we won't get great URLs, but it'll be fast.

Not only would this be a terrific flow, it would be a great real-world example of Pulumi in action.

**Comment by chrsmith** [comment]
For the immediate term, I was thinking we'd just hook up Netlify so we can have something to unblock people while we work on revamping our documentation.

Having a "Pulumi-powered Review App" would be the ideal end-result though. It should be very doable using GitHub Actions, I think it's just a matter of getting the workflow right. For reference https://github.com/pulumi/actions/issues/9 is the tracking this type of feature in that repo, and is linked to some other PRs that have been sent out around enabling that type of thing.

**Comment by chrsmith** [comment]
Christian created a solution for this internally. We'll still want to create a Pulumi stack that can provide this service, and using GitHub Actions would be an ideal compliment. But that effort is out of scope for the current milestone.
